pkg/pipeline/conntrack/hash_test.go

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package conntrack
+
+import (
+	"hash/fnv"
+	"testing"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/stretchr/testify/require"
+)
+
+func NewFlowLog(srcIP string, srcPort int, dstIP string, dstPort int, protocol int, bytes, packets int) config.GenericMap {
+	return config.GenericMap{
+		"SrcAddr": srcIP,
+		"SrcPort": srcPort,
+		"DstAddr": dstIP,
+		"DstPort": dstPort,
+		"Proto":   protocol,
+		"Bytes":   bytes,
+		"Packets": packets,
+	}
+}
+
+var hasher = fnv.New32a()
+
+func TestComputeHash_Unidirectional(t *testing.T) {
+	keyDefinition := api.KeyDefinition{
+		FieldGroups: []api.FieldGroup{
+			{
+				Name: "src",
+				Fields: []string{
+					"SrcAddr",
+					"SrcPort",
+				},
+			},
+			{
+				Name: "dst",
+				Fields: []string{
+					"DstAddr",
+					"DstPort",
```

Comment:
If you include the ephemeral port in the hash, it will count a lot of connections.  We need to get an agreement on what a "connection" is.  Example: Let's say you access a web page.  Is that one connection?  With this implementation, it could be anywhere from 1 to 6 connections.  
  
A typical web page will refer to JavaScript files, CSS files, images, etc.  The browser will need to fetch these files.  Pretty much all modern browsers today will allow up to 6 simultaneous connections to the same domain, and will reuse these connections to fetch all the files.  Each of these connections will have a different ephemeral port.

Reply:
I see. I tried to make the decision of what is a connection configurable. But it's still not flexible enough to support this use-case.

This specific unit test configures the classic 5-tuple to distinguish between connections. The other unit test uses a slightly different configuration. It uses the same 5-tuple but includes both flow directions in the same connection.